HAL.md

```diff
@@ -0,0 +1,164 @@
+TA-endpoint and HAL overview
+===
+
+# Abstract 
+The TA-endpoint uses HAL(hardware abstract layer) to encapsulate the logic of driver of kinds of device. HAL layer support such as uart, GPIO, secure storage and other low-level operators used in TA-endpoint. The TA-endpoint is a daemon which base on Linux which offer monitor for device and can send the message to tangle-accelerator. HAL layer lets user only needs to implement their device operators from abstract interface which defined inside `device.h` but not need to know how TA-endpoint work. 
```

Comment:
Change the first line to
> Hardware Abstraction Layer (HAL) defines a standard interface for hardware vendors to implement, which enables Tangle-Accelerator to be aware of lower-level driver implementations.

Don't mention TA-endpoint anymore since we are going to consolidate.